web/src/main/java/nl/tehdreamteam/se42/web/token/generator/TokenGenerator.java

```diff
@@ -0,0 +1,33 @@
+package nl.tehdreamteam.se42.web.token.generator;
+
+import nl.tehdreamteam.se42.domain.User;
+import nl.tehdreamteam.se42.web.token.Token;
+import nl.tehdreamteam.se42.web.token.generator.id.TokenIdGenerator;
+
+/**
+ * A {@code TokenGenerator} provides the strategy to generate {@link Token Tokens}.
+ */
+public interface TokenGenerator {
+
+    /**
+     * Generates a {@link Token} for the given {@link User}.
+     *
+     * @return A generated {@code Token} for the given {@code user}.
+     */
+    Token generate(User user);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Document the parameter(s): user [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AUndocumentedApi)